### PR TITLE
Feature/299 backdrop

### PR DESCRIPTION
**Pull request by eirikbacker**
- Fixes coreDialog (and other components) in Edge (where custom-element polyfill and toggleAttribute polyfill did not play well together. Changes `toggleAttribute` to ponyfill, utilising already well known and patched functions such as `setAttribute` https://github.com/webcomponents/custom-elements/blob/master/src/Patch/Element.js#L134)
- Breaking change; moving away from confusing `modal` concept (is modal with or without backdrop? is it same as `aria-modal`? etc.)
- Instead introducing `backdrop` attribute: `true` | `false` | `ElementID` (for targeting custom backdrop element - nice to have when doing server side rendering)
- Listens to changes in `backdrop` prop (setting `.backdrop = false` hides backdrop)

**Before:**
```
<core-dialog></core-dialog>                      <!-- with backdrop -->
<core-dialog modal="false"></core-dialog>        <!-- no backdrop -->
<core-dialog></core-dialog><backdrop></backdrop> <!-- custom backdrop -->
```

**After:**
```
<core-dialog></core-dialog>                                  <!-- with backdrop -->
<core-dialog backdrop="false"></core-dialog>                 <!-- no backdrop -->
<core-dialog backdrop="my"></core-dialog><div id="my"></div> <!-- custom backdrop -->
```

### CODE REVIEW

File: packages/utils.js [hunk 1]
```diff
@@ -117,9 +109,21 @@ export function throttle (callback, ms) {
   }
 }
 
+/**
+ * toggleAttribute (Ponyfill for IE and Edge, fixes #299)
+ * @param {Element} el  The new throttled function
+ * @param {String} name The name of the attribute to be toggled
+ * @param {Boolean} force Determine whether the attribute should be added or removed
```
Comment (laat) [inline packages/utils.js]: Jeg skjønner ikke hva `force` gjør

File: packages/utils.js [hunk 2]
```diff
@@ -117,9 +109,21 @@ export function throttle (callback, ms) {
   }
 }
 
+/**
+ * toggleAttribute (Ponyfill for IE and Edge, fixes #299)
+ * @param {Element} el  The new throttled function
```
Comment (laat) [inline packages/utils.js]: `The new throttled function` kan fjernes

File: packages/utils.js (hunk 1 above)
Comment (eirikbacker) [reply packages/utils.js]: Godt poeng, var for å være i tråd med https://developer.mozilla.org/en-US/docs/Web/API/Element/toggleAttribute :)

File: packages/utils.js (hunk 2 above)
Comment (eirikbacker) [reply packages/utils.js]: hehe, takk, typo :)

File: packages/utils.js (hunk 1 above)
Comment (eirikbacker) [reply packages/utils.js]: Will continue using `force` as `value` is misleading. Changed description and added link to original docs: https://developer.mozilla.org/en-US/docs/Web/API/Element/toggleAttribute.
Also see `force` used as parameter name in `.toggle` here https://developer.mozilla.org/en-US/docs/Web/API/Element/classList :)

**Comment by htor** [comment]
nice, looks much nicer without `modal`!  

but it think there's inconsistency in the way the attribute works: 
* initializing with `<core-dialog backdrop="false">` will not insert a backdrop element in the dom
* initializing with `<core-dialog backdrop="true"> will create a backdrop element but subsequent changes to the `backdrop` attr will only hide/unhide the backdrop (with `hidden`), not remove it from the dom. 
shouldn't we insert/remove it from the dom on changes?

**Comment by eirikbacker** [comment]
> subsequent changes to the `backdrop` attr will only hide/unhide the backdrop (with `hidden`), not remove it from the dom.

Sure, good point - can be done :) On holiday today, but will push fix on Monday :)

**Comment by esseb** [comment]
The `backdrop` prop feels weird in the React version. I keep trying to do `<CoreDialog backdrop={false} ...>` which doesn't work. Maybe `@nrk/custom-element-to-react` could map boolean string attributes to be `true/false/string` props in React?

Besides that, this branch works great. No issues in Edge any more 🎉

**Comment by eirikbacker** [comment]
@esseb Good point! Added `backdrop` to `@nrk/custom-element-to-react` props so this works now :)

**Pull request merged**